lib/screens/questionaire_screen/widgets/cover.dart

```diff
@@ -56,68 +60,149 @@ class Cover extends StatelessWidget {
+            right: 20,
+            child: Image(
+              image: AssetImage('assets/Tipp.png'),
+              height: 50,
+              width: 50,
             ),
-            if (backFunction == null)
-              PicosInkWellButton(
-                text: start,
-                onTap: nextFunction ?? () {},
-              )
-            else
-              PicosAddButtonBar(
-                shadows: false,
-                leftButton: PicosInkWellButton(
-                  padding: const EdgeInsets.only(
-                    left: 30,
-                    right: 13,
-                    top: 15,
-                    bottom: 10,
-                  ),
-                  text: back,
-                  onTap: backFunction ?? () {},
-                  buttonColor1: theme.grey3,
-                  buttonColor2: theme.grey1,
-                ),
-                rightButton: PicosInkWellButton(
-                  padding: const EdgeInsets.only(
-                    right: 30,
-                    left: 13,
-                    top: 15,
-                    bottom: 10,
+          ),
+        ],
+      ),
+    );
+
+    if (backFunction == null && nextFunction != null) {
+      bottomWidget = PicosInkWellButton(
+        text: start,
+        onTap: nextFunction!,
+      );
+      sizedBoxHeight = height / 5;
+      fontSize = 40;
+    } else if (backFunction != null || nextFunction != null) {
+      bottomWidget = PicosAddButtonBar(
+        shadows: false,
+        leftButton: PicosInkWellButton(
+          padding: const EdgeInsets.only(
+            left: 30,
+            right: 13,
+            top: 15,
+            bottom: 10,
+          ),
+          text: back,
+          onTap: backFunction!,
+          buttonColor1: theme.grey3,
+          buttonColor2: theme.grey1,
+        ),
+        rightButton: PicosInkWellButton(
+          padding: const EdgeInsets.only(
+            right: 30,
+            left: 13,
+            top: 15,
+            bottom: 10,
+          ),
+          text: next,
+          onTap: nextFunction!,
+        ),
+      );
+      sizedBoxHeight = height / 5;
+      fontSize = 40;
+    }
+
+    Widget picosBody = PicosBody(
+      child: Column(
+        children: <Widget>[
+          SizedBox(
+            height: height / 8,
+          ),
+          Image(
+            image: AssetImage(image),
+          ),
+          Padding(
+            padding: const EdgeInsets.only(top: 60),
+            child: Text(
+              title,
+              textAlign: TextAlign.center,
+              style: TextStyle(
+                fontSize: fontSize,
+                fontWeight: FontWeight.bold,
+                color: Colors.white,
+              ),
+            ),
+          ),
+          SizedBox(
+            height: sizedBoxHeight,
+          ),
+          bottomWidget,
+        ],
+      ),
+    );
+
+    if (finishCover) {
+      picosBody = GestureDetector(
+        onTap: () => Navigator.of(context).pop(),
```

Comment:
This solution seems a bit complicated to me.
Why don't you just make the GestureDetector a standard part of the cover and set the onTap function based on the conditions you already have?

Reply:
I've got another solution for that.